src/localforage.js

```diff
@@ -306,36 +298,74 @@
         setDriver(drivers, callback, errorCallback) {
             var self = this;
 
-            if (typeof drivers === 'string') {
+            if (!isArray(drivers)) {
                 drivers = [drivers];
             }
 
-            this._driverSet = new Promise(function(resolve, reject) {
-                var driverName = self._getFirstSupportedDriver(drivers);
-                var error = new Error('No available storage method found.');
+            var supportedDrivers = this._getSupportedDrivers(drivers);
 
-                if (!driverName) {
-                    self._driverSet = Promise.reject(error);
-                    reject(error);
-                    return;
-                }
+            function setDriverToConfig() {
+                self._config.driver = self.driver();
+            }
 
-                self._dbInfo = null;
-                self._ready = null;
+            function initDriver(supportedDrivers) {
+                return function() {
+                    var crntDriverPos = 0;
```

Comment:
I'm going to rename this to `currentDriverPosition`; we have the bytes to spare :wink: